libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseP_CAN.cpp

```diff
@@ -0,0 +1,58 @@
+#include "AP_RangeFinder_TOFSenseP_CAN.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+#if AP_RANGEFINDER_TOFSENSEP_CAN_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+
+RangeFinder_MultiCAN *AP_RangeFinder_TOFSenseP_CAN::multican_TOFSenseP;
+
+/*
+  constructor
+ */
+AP_RangeFinder_TOFSenseP_CAN::AP_RangeFinder_TOFSenseP_CAN(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    AP_RangeFinder_Backend_CAN(_state, _params)
+{
+    if (multican_TOFSenseP == nullptr) {
+        multican_TOFSenseP = new RangeFinder_MultiCAN(AP_CAN::Protocol::TOFSenseP, "TOFSenseP MultiCAN");
+        if (multican_TOFSenseP == nullptr) {
+            AP_BoardConfig::allocation_error("Rangefinder_MultiCAN");
+        }
+    }
+
+    {
+        // add to linked list of drivers
+        WITH_SEMAPHORE(multican_TOFSenseP->sem);
```

Comment:
Maybe we need a nullptr here check as well.. essentially the same question as the other driver.